components/core/src/lib.rs

```diff
@@ -47,3 +47,9 @@ impl ChannelIdent {
 impl fmt::Display for ChannelIdent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { write!(f, "{}", self.0) }
 }
+
+pub fn channel_from_env() -> Option<ChannelIdent> {
+    env::var(ChannelIdent::ENVVAR).ok().map(ChannelIdent::from)
+}
+
+pub fn default_channel() -> ChannelIdent { channel_from_env().unwrap_or_default() }
```

Comment:
You don't actually need `default_channel` or `channel_from_env`, since the underlying `env_config!` infrastructure provides this functionality for you via `ChannelIdent::configured_value()` (indeed, this is the precise usecase that macro was originally written to serve!). Let me know if you want to talk more about that :smile: 

As an aside, and assuming that macro wasn't present, it would have been good to associate these functions with the type directly, allowing you to call them like `ChannelIdent::channel_from_env()`. That would be even better as `ChannelIdent::from_env()`, since you can let the type "do some of the talking" for you. That same reasoning would also have led to `ChannelIdent::default()`, which in turn would have led to some further thought about whether that should have been an implementation of `Default`, rather than a function in the `ChannelIdent` impl itself (In this case, it would _not_ have been correct to make this a `Default` implementation, because `Default` implementations that depend on the environment are a very sneaky kind of evil... and that's why `env_config!` creates a function called `configured_value` to do this.)

Also, it's a good idea to not make things public that don't need to be. `channel_from_env` was only ever called from `default_channel`, so it could have just been private (if not completely inlined with `default_channel`'s implementation). The Rust compiler is really good at alerting you to unused code, but you have to make sure your visibility settings are correct. If you say an item is public, the compiler takes you at your word, and assumes that external clients of your library could call them. On the other hand, if they're not public, and the compiler sees there are no callers, it'll tell you and you get the joy of deleting code :tada: Don't deprive yourself of that joy :rofl:

Reply:
Thank you! I updated to using the macro and removing `default_channel` and `channel_from_env` in https://github.com/habitat-sh/habitat/pull/7215/commits/1b608543359688820c9848f19e924a778f3f7f86